applications/TrilinosApplication/custom_strategies/builder_and_solvers/trilinos_block_builder_and_solver.h

```diff
@@ -4,954 +4,830 @@
+
+            // assemble the elemental contribution
+            TSparseSpace::AssembleLHS(rA, LHS_Contribution, equation_ids_vector);
         }
 
-        //finalizing the assembly
-        A.GlobalAssemble();
+        // finalizing the assembly
+        rA.GlobalAssemble();
         KRATOS_CATCH("")
-
     }
 
+    /**
+     * @brief Build a rectangular matrix of size n*N where "n" is the number of
+     * unrestrained degrees of freedom and "N" is the total number of degrees of
+     * freedom involved.
+     * @details This matrix is obtained by building the total matrix without the
+     * lines corresponding to the fixed degrees of freedom (but keeping the
+     * columns!!)
+     * @param pScheme The integration scheme considered
+     * @param rModelPart The model part of the problem to solve
+     * @param A The LHS matrix
+     */
+    void BuildLHS_CompleteOnFreeRows(typename TSchemeType::Pointer pScheme,
+                                     ModelPart& rModelPart,
+                                     TSystemMatrixType& A) override
+    {
+        KRATOS_ERROR << "Method BuildLHS_CompleteOnFreeRows not implemented in "
+                        "Trilinos Builder And Solver"
+                     << std::endl;
+    }
 
-
-    //**************************************************************************
-    //**************************************************************************
-    /** Solve the linear problem.
+    /**
+     * @brief This is a call to the linear system solver
+     * @param A The LHS matrix
+     * @param Dx The Unknowns vector
+     * @param b The RHS vector
      */
-    void SystemSolveWithPhysics(
-        TSystemMatrixType& A,
-        TSystemVectorType& Dx,
-        TSystemVectorType& b,
-        ModelPart& r_model_part
-    )
+    void SystemSolveWithPhysics(TSystemMatrixType& rA,
+                                TSystemVectorType& rDx,
+                                TSystemVectorType& rb,
+                                ModelPart& rModelPart)
     {
         KRATOS_TRY
 
         double norm_b;
-        if(TSparseSpace::Size(b) != 0)
-            norm_b = TSparseSpace::TwoNorm(b);
+        if (TSparseSpace::Size(rb) != 0)
+            norm_b = TSparseSpace::TwoNorm(rb);
         else
             norm_b = 0.00;
 
-        if(norm_b != 0.00)
-        {
-            if (this->GetEchoLevel()>1)
-                if (mrComm.MyPID() == 0) KRATOS_WATCH("entering in the solver");
-
-            if(BaseType::mpLinearSystemSolver->AdditionalPhysicalDataIsNeeded() )
-                BaseType::mpLinearSystemSolver->ProvideAdditionalData(A, Dx, b, BaseType::mDofSet, r_model_part);
-
-            if (this->GetEchoLevel()>3)
-            {
-                EpetraExt::RowMatrixToMatrixMarketFile( "A.mm", A, "matrixA", "lhs_matrix", true);
-                EpetraExt::MultiVectorToMatrixMarketFile( "b.mm", b, "vectorb","rhs_vector",true);
```

Comment:
please don't remove this

Reply:
I thought the strategy should outpu this and not the BuilderAndSolver ...

Reply:
the normal b&s does not have this.

Reply:
ah true, sorry for the noise